common/maven-url-handler/src/main/java/io/helidon/build/url/mvn/Handler.java

```diff
@@ -0,0 +1,498 @@
+
+                artifactId = segments[1];
+                checkStringHealth(artifactId, "artifactId");
+            } else {
+                throw new MalformedURLException("Missing element in maven URL. Syntax " + SYNTAX);
+            }
+
+            switch (segments.length) {
+                case 3:
+                    parseNoClassifierOrType(segments);
+                    break;
+                case 4:
+                    parseWithClassifierOrType(segments);
+                    break;
+                case 5:
+                    parseCompleteUrl(segments);
+                    break;
+                default:
+                    throw new MalformedURLException("Invalid path. Syntax " + SYNTAX);
+            }
+        }
+
+        private void parseCompleteUrl(String[] segments) throws MalformedURLException {
+            if (segments.length != 5) {
+                throw new MalformedURLException("Invalid. Syntax " + SYNTAX);
+            }
+            version = segments[2];
+            checkStringHealth(version, "version");
+
+            classifier = segments[3];
+            checkStringHealth(classifier, "classifier");
+
+            segments = segments[4].split(FILE_SEPARATOR);
+
+            type = checkTypeHealth(segments[0]) ? segments[0] : DEFAULT_TYPE;
+            checkStringHealth(type, "type");
+
+            buildPathFromArchive(segments);
+        }
+
+        private void parseWithClassifierOrType(String[] segments) throws MalformedURLException {
+            if (segments.length != 4) {
+                throw new MalformedURLException("Invalid. Syntax " + SYNTAX);
+            }
+
+            version = segments[2];
+            checkStringHealth(version, "version");
+
+            segments = segments[3].split(FILE_SEPARATOR);
+
+            if (checkTypeHealth(segments[0])) {
+                type = segments[0];
+            } else {
+                classifier = segments[0];
+            }
+
+            buildPathFromArchive(segments);
+        }
+
+        private void parseNoClassifierOrType(String[] segments) throws MalformedURLException {
+            if (segments.length != 3) {
+                throw new MalformedURLException("Invalid. Syntax " + SYNTAX);
+            }
+            segments = segments[2].split(FILE_SEPARATOR);
+
+            version = segments[0];
+            checkStringHealth(version, "version");
+
+            buildPathFromArchive(segments);
+        }
+
+        private void buildPathFromArchive(String[] segments) {
+            StringBuilder builder = new StringBuilder();
+            for (int i = 1; i < segments.length - 1; i++) {
+                builder.append(segments[i]).append("/");
+            }
+            builder.append(segments[segments.length - 1]);
+            pathFromArchive = builder.toString();
+        }
+
+        private void checkStringHealth(String patient, String patientName) throws MalformedURLException {
+            if (patient.trim().length() == 0) {
```

Comment:
The health theme here doesn't have much value ;)
Can you rename this to something else like `validateString` or `checkString`, it can be static too.